src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java

```diff
@@ -49,36 +49,34 @@ public FtpClient(
     }
     // endregion
 
-    public void upload(FileToSend file, String serviceFolder) {
-        Instant start = Instant.now();
+    public void upload(FileToSend file, String serviceFolder, SFTPClient sftpClient) {
+        String folder = file.isSmokeTest
+            ? configProperties.getSmokeTestTargetFolder()
+            : String.join("/", configProperties.getTargetFolder(), serviceFolder);
 
-        runWith(sftp -> {
-            boolean isSuccess = false;
-
-            try {
-                String folder = file.isSmokeTest
-                    ? configProperties.getSmokeTestTargetFolder()
-                    : String.join("/", configProperties.getTargetFolder(), serviceFolder);
+        String path = String.join("/", folder, file.getName());
+        Instant start = Instant.now();
+        boolean isSuccess = false;
 
-                String path = String.join("/", folder, file.getName());
-                sftp.getFileTransfer().upload(file, path);
+        try {
+            sftpClient.getFileTransfer().upload(file, path);
 
-                logger.info(
-                    "File uploaded. Time: {}, Size: {}, Folder: {}",
-                    ChronoUnit.MILLIS.between(start, Instant.now()) + "ms",
-                    file.content.length / 1024 + "KB",
-                    serviceFolder
-                );
+            isSuccess = true;
+        } catch (IOException exc) {
+            throw new FtpException("Unable to upload file.", exc);
+        } finally {
+            long millis = ChronoUnit.MILLIS.between(start, Instant.now());
```

Comment:
Nice calculating it only once 👍

Reply:
info on successful upload (below) is now logged even on failures 👎